service/history/execution/mutable_state_builder.go

```diff
@@ -4753,6 +4781,15 @@ func (e *mutableStateBuilder) checkMutability(
 	return nil
 }
 
+func (e *mutableStateBuilder) insertWorkflowRequest(request persistence.WorkflowRequest) {
+	if e.domainEntry != nil && e.config.EnableStrongIdempotency(e.domainEntry.GetInfo().Name) && request.RequestID != "" {
+		if _, ok := e.workflowRequests[request]; ok {
+			e.logWarn("error encountering duplicate request", tag.WorkflowRequestID(request.RequestID))
```

Comment:
do we have the workflow and domain info there by any chance?

Reply:
we do